### PR TITLE
Update ESM v2 status

### PR DESCRIPTION
**Pull request by joe4dev**
Update status for 3.8 release

Related to https://github.com/localstack/localstack/pull/11625

### CODE REVIEW

File: content/en/user-guide/aws/lambda/index.md [hunk 1]
```diff
@@ -180,25 +180,20 @@ The following event sources are supported in LocalStack:
...
-
-The limitations compared to ESM v1 include:
-- Lambda Failure Destinations do not yet support FIFO SQS queues and SNS topics.
-- Lambda Failure Destination Messages do not yet match the AWS structure, using a `context` similar to EventBridge Pipes instead of `requestContext` and `responseContext`.
-- Partial Batch Responses using `FunctionResponseTypes` are not yet fully supported.
-- Managed Streaming for Apache Kafka (MSK) event source is not yet supported.
-- ESM Lifecycle State Updates only provide basic support for state updates, such as no failure states, and `LastProcessingResult` is not consistently updated.
-- Persistence is not yet supported
+- Improved AWS parity, for example related to filtering, SQS polling, and (partial) batch failure handling.
+- Improved Kafka support for Amazon MSK and Self-Managed Kafka.
```
Comment (gregfurman) [inline content/en/user-guide/aws/lambda/index.md:192]: Should we mention limitations i.e currently no partial batch failure handling for Kafka or should this rather be a blog post?

File: content/en/user-guide/aws/lambda/index.md [hunk 2]
```diff
@@ -180,25 +180,20 @@ The following event sources are supported in LocalStack:
...
-- Improved AWS parity, for example related to filtering and SQS polling.
-
-The limitations compared to ESM v1 include:
-- Lambda Failure Destinations do not yet support FIFO SQS queues and SNS topics.
-- Lambda Failure Destination Messages do not yet match the AWS structure, using a `context` similar to EventBridge Pipes instead of `requestContext` and `responseContext`.
-- Partial Batch Responses using `FunctionResponseTypes` are not yet fully supported.
-- Managed Streaming for Apache Kafka (MSK) event source is not yet supported.
-- ESM Lifecycle State Updates only provide basic support for state updates, such as no failure states, and `LastProcessingResult` is not consistently updated.
-- Persistence is not yet supported
+- Improved AWS parity, for example related to filtering, SQS polling, and (partial) batch failure handling.
```
Comment (gregfurman) [inline content/en/user-guide/aws/lambda/index.md:191]:
```suggestion
- Improved AWS parity, related to events filtering, configuring failure destinations, (partial) batch failure handling, and SQS, Kinesis, & DynamoDB Streams events polling.
```

**Review by gregfurman — APPROVED** [review]
LGTM 👍  Just a question surrounding documenting (current) limitations of Kafka and an optional phrasing adjustment.

File: content/en/user-guide/aws/lambda/index.md (hunk 1 above)
Comment (joe4dev) [reply content/en/user-guide/aws/lambda/index.md:192]: I think mentioning limitations makes sense in the docs. Feel free to add anything.
Ultimately, the docs could mimic the high-level features similar to the AWS docs and indicate what's supported and what not clearly.

**Review by HarshCasper — APPROVED** [review]
Merging this for the release

**Pull request merged**